test/controllers/EntryDetailsControllerSpec.scala

```diff
@@ -79,31 +90,88 @@ class EntryDetailsControllerSpec extends ControllerSpecBase {
         redirectLocation(result) mustBe Some(controllers.routes.AcceptanceDateController.onLoad().url)
       }
 
       "return a SEE OTHER response and redirect to correct location for date AFTER EU exit" in new Test {
-        private val request = fakeRequest.withFormUrlEncodedBody(buildForm(day=Some("02"),month=Some("01"),year=Some("2021")):_*)
+        private val request = fakeRequest.withFormUrlEncodedBody(buildForm(day = Some("02"), month = Some("01"), year = Some("2021")): _*)
         lazy val result: Future[Result] = controller.onSubmit(request)
         status(result) mustBe Status.SEE_OTHER
         redirectLocation(result) mustBe Some(controllers.routes.AcceptanceDateController.onLoad().url)
       }
 
       "update the UserAnswers in session" in new Test {
-        private val request = fakeRequest.withFormUrlEncodedBody(buildForm():_*)
+        private val request = fakeRequest.withFormUrlEncodedBody(buildForm(): _*)
         await(controller.onSubmit(request))
         verifyCalls()
       }
     }
 
+    "not in change mode after successful submission" should {
+      "not in change mode after successful submission redirect to Acceptance date page" in new Test {
+        private val request = fakeRequest.withFormUrlEncodedBody(buildForm(day = Some("02"), month = Some("01"), year = Some("2021")): _*)
+        override val userAnswers: Option[UserAnswers] =
+          Some(UserAnswers("some-cred-id")
+            .set(EntryDetailsPage, EntryDetails("123", "123456Q", LocalDate.now)).success.value
+            .set(CheckModePage, false).success.value
+          )
+        lazy val result: Future[Result] = controller.onSubmit(request)
+        status(result) mustBe Status.SEE_OTHER
+        redirectLocation(result) mustBe Some(controllers.routes.AcceptanceDateController.onLoad().url)
+      }
+    }
+
+    "in change mode after successful submission" should {
+      "redirect to Check your answers page" in new Test {
+        private val request = fakeRequest.withFormUrlEncodedBody(buildForm(day = Some("02"), month = Some("01"), year = Some("2021")): _*)
+        override val userAnswers: Option[UserAnswers] =
+          Some(UserAnswers("some-cred-id")
+            .set(EntryDetailsPage, EntryDetails("123", "123456Q", LocalDate.now)).success.value
+            .set(CheckModePage, true).success.value
+          )
+        lazy val result: Future[Result] = controller.onSubmit(request)
+        status(result) mustBe Status.SEE_OTHER
+        redirectLocation(result) mustBe Some(controllers.routes.CheckYourAnswersController.onLoad().url)
+      }
+    }
+
     "payload contains invalid data" should {
       "return a BAD REQUEST" in new Test {
         val result: Future[Result] = controller.onSubmit(fakeRequest)
         status(result) mustBe Status.BAD_REQUEST
       }
     }
   }
+
+  "backLink" when {
+
+    "not in change mode" should {
+      "when loading page back button should take you to Acceptance date page" in new Test {
```

Comment:
Suggest re-phrasing as "point to Acceptance date page" so that the test output will read better:

backLink when
  not in change mode should
    point to Acceptance date page

Same for below too please.